src/charm.py

```diff
@@ -110,43 +114,47 @@ def _on_webui_pebble_ready(self, event: EventBase) -> None:
             return
         if not self._container.can_connect():
             self.unit.status = WaitingStatus("Waiting for container to be ready")
-            event.defer()
             return
-        if not self._config_file_is_written():
+        if not self._container.exists(path=BASE_CONFIG_PATH):
```

Comment:
What happens if the database event gets consumed by this path?

Reply:
What I am afraid of is that we would be blocked because even if `_configure` is re executed under any other event, the config file will never be written unless there is another Database event

Reply:
Yes, that is what I fear too

Reply:
so I think that keeping those defers make sense. I pushed a new version
good catch :)